src/ucs/sys/sys.c

```diff
@@ -330,25 +330,41 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
-size_t ucs_get_huge_page_size()
+size_t ucs_get_meminfo_entry(char *pattern)
 {
-    static size_t huge_page_size = 0;
     char buf[256];
-    int size_kb;
+    size_t val = 0;
     FILE *f;
+    f = fopen("/proc/meminfo", "r");
+    if (f != NULL) {
+        while (fgets(buf, sizeof(buf), f)) {
+            if (sscanf(buf, pattern, &val) == 1) {
+                break;
+            }
+        }
+        fclose(f);
+    }
+    return val;
+}
+
+size_t ucs_get_memfree_size()
+{
+    static size_t memfree_size = 0;
```

Comment:
need to avoid this - the amount of free memory can change during the course of the program